pkg/resources/management.cattle.io/v3/project/validator_test.go

```diff
@@ -929,6 +930,417 @@ func TestProjectValidation(t *testing.T) {
+				},
+				Spec: v3.ProjectSpec{
+					ClusterName: "testcluster",
+					ContainerDefaultResourceLimit: &v3.ContainerResourceLimit{
+						RequestsCPU:    "1m",
+						LimitsCPU:      "20m",
+						RequestsMemory: "1Mi",
+						LimitsMemory:   "-20Mi",
+					},
+				},
+			},
+		},
+		{
+			name: "invalid value on container cpu request causes an error",
+			newProject: &v3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "testcluster",
+				},
+				Spec: v3.ProjectSpec{
+					ClusterName: "testcluster",
+					ContainerDefaultResourceLimit: &v3.ContainerResourceLimit{
+						RequestsCPU:    "apple",
+						LimitsCPU:      "20m",
+						RequestsMemory: "1Mi",
+						LimitsMemory:   "20Mi",
+					},
+				},
+			},
+		},
+		{
+			name: "invalid value on container cpu limit causes an error",
+			newProject: &v3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "testcluster",
+				},
+				Spec: v3.ProjectSpec{
+					ClusterName: "testcluster",
+					ContainerDefaultResourceLimit: &v3.ContainerResourceLimit{
+						RequestsCPU:    "1m",
+						LimitsCPU:      "apple",
+						RequestsMemory: "1Mi",
+						LimitsMemory:   "20Mi",
+					},
+				},
+			},
+		},
+		{
+			name: "invalid value on container memory request causes an error",
+			newProject: &v3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "testcluster",
+				},
+				Spec: v3.ProjectSpec{
+					ClusterName: "testcluster",
+					ContainerDefaultResourceLimit: &v3.ContainerResourceLimit{
+						RequestsCPU:    "1m",
+						LimitsCPU:      "20m",
+						RequestsMemory: "apple",
+						LimitsMemory:   "20Mi",
+					},
+				},
+			},
+		},
+		{
+			name: "invalid value on container memory limit causes an error",
+			newProject: &v3.Project{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "testcluster",
+				},
+				Spec: v3.ProjectSpec{
+					ClusterName: "testcluster",
+					ContainerDefaultResourceLimit: &v3.ContainerResourceLimit{
+						RequestsCPU:    "1m",
+						LimitsCPU:      "20m",
+						RequestsMemory: "1Mi",
+						LimitsMemory:   "apple",
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		for _, operation := range []admissionv1.Operation{admissionv1.Create, admissionv1.Update} {
+			test.operation = operation
+			name := fmt.Sprintf("%s on %s", test.name, strings.ToLower(string(test.operation)))
+			t.Run(name, func(t *testing.T) {
+				t.Parallel()
+				ctrl := gomock.NewController(t)
+				state := testState{
+					clusterCache: fake.NewMockNonNamespacedCacheInterface[*v3.Cluster](ctrl),
+				}
+				if test.operation == admissionv1.Create {
+					state.clusterCache.EXPECT().Get("testcluster").Return(&v3.Cluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "testcluster",
+						},
+					}, nil)
+				}
+				if test.operation == admissionv1.Update {
+					test.oldProject = &v3.Project{
```

Comment:
Nit:
Since you don't actually specify the old project in your tests, you don't need to add the project as a member variable on the test. This has the additional benefit of just allowing you to have a var on the test called `test.Project` instead of only "new" project.